components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerSnapshotTests.m

```diff
@@ -0,0 +1,213 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY MDCTextFieldSnapshotTestsIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCSnapshotTestCase.h"
+#import "MDCTextFieldSnapshotTestsStrings.h"
+#import "MaterialTextFields.h"
+
+@interface MDCTextFieldOutlinedControllerSnapshotTests : MDCSnapshotTestCase
+@property(nonatomic, strong) MDCTextField *textField;
+@property(nonatomic, strong) MDCTextInputControllerOutlined *textFieldController;
+@end
+
+@implementation MDCTextFieldOutlinedControllerSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.textField = [[MDCTextField alloc] init];
+  self.textFieldController =
+      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];
+}
+
+- (void)tearDown {
+  self.textFieldController = nil;
+  self.textField = nil;
+
+  [super tearDown];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Helpers
+
+- (void)triggerTextFieldLayout {
+  CGSize aSize = [self.textField sizeThatFits:CGSizeMake(300, INFINITY)];
+  self.textField.bounds = CGRectMake(0, 0, aSize.width, aSize.height);
+  [self.textField layoutIfNeeded];
+}
+
+- (void)generateSnapshotAndVerify {
+  [self triggerTextFieldLayout];
+
+  self.testView = [self addBackgroundViewToView:self.textField];
+}
```

Comment:
This method doesn't appear to verify the snapshots which caught me by surprise. I've filed https://github.com/material-components/material-components-ios/issues/5941 to track making the snapshot tests more explicit.

Reply:
It's handled by the snapshot testing helper/framework methods. Since we want to keep one image per test, it was an effective way to enforce that expectation.